Potapov/telegram bot

ptpchh opened the pull request:
пофиксил ошибки и недочёты, щас вроде всё должно быть хорошо. единственное, кривовато сделано добавление предметов на старте, потому что всё ещё нет баз данных)

ptpchh committed:
Potapov/tg_bot/ver_1.2
fixed status reduction + move files for easier understanding

ptpchh committed:
Potapov/tg_bot/ver_1.3
rework commands, now user don't need to remember his and his pet IDs, 'cause they made from ChatId. also other minor fixes

ptpchh committed:
Potapov/tg_bot/ver_1.4
rework placement of functions. now, it's easier to read a code

ptpchh committed:
Potapov/tg_bot/ver_1.5
added awful, but working functions for starting items (while there is no DB

lostflydev commented on src/main/java/angryDragon/business/addition/PackOfItems.java:
```diff
@@ -0,0 +1,75 @@
+    public PackOfItems(){
+        List<Item> items = new ArrayList<>();
+
+        Item potato = new Item("I13853", "Potato", 7, HUNGER, 10);
```
это зачем

ptpchh replied:
забыл убрать

lostflydev commented on src/main/java/angryDragon/business/domain/status/Status.java:
```diff
@@ -24,14 +24,14 @@ public int getJoy(){
 
     public void setEnergy(int valueOfEnergy){
-        this.energy = Math.min(valueOfEnergy, 100);
+        this.energy = Math.max(Math.min(valueOfEnergy, 100), 0);
```
так вы пытаетесь избежать что valueOfEnergy отрицательно?

ptpchh replied:
да, так создаются минимальные и максимальные границы

lostflydev commented on src/main/java/angryDragon/business/handlers/tgApiHandler/TgApiHandler.java:
```diff
@@ -0,0 +1,316 @@
+            return "Диалог отменён.";
+        }
+
+        return switch (session.getState()) {
```
в ретурне такую большую функцию нехорошо

ptpchh replied:
но у тебя у самого так в коде было)
ладно, тогда что сделать? выделить свитч в отдельную переменную и делать return именно этой переменной?

ptpchh replied:
мне сама интелия предлагает объединить это в return

lostflydev commented on src/main/java/angryDragon/business/handlers/userSession/KeyPair.java:
```diff
@@ -0,0 +1,26 @@
+
+import java.util.Objects;
+
+public class KeyPair {
```
для чего нужен?

ptpchh replied:
для того, чтобы можно было верно сохранять данные (временные) и переносить их. мало ли два пользователя будут одновременно добавлять предмет, тогда может произойти коллизия

lostflydev commented on src/main/java/angryDragon/business/repository/impl/AllExistingItemsRepositoryImpl.java:
```diff
@@ -34,4 +39,57 @@ public Set<Item> getAllExistingItems(){
 
+    @Override
+    public void packOfItems(){
+        Item potato = new Item("I13853", "Potato", 7, HUNGER, 10);
```
wtf)

ptpchh committed:
Potapov/tg_bot/ver_1.6
minor changes and fixes

lostflydev commented on src/main/java/angryDragon/business/handlers/tgApiHandler/TgApiHandler.java:
```diff
@@ -43,9 +40,25 @@ public TgApiHandler() {
+                String[] parts = newItem.split("[,\\s]+");
+                repositoryComponent.getAllExistingItemsRepository().addItem(new Item(parts[0], parts[1],
+                        Integer.parseInt(parts[2]), WhatItemRestore.valueOf(parts[3]), Integer.parseInt(parts[4])));
+                serviceComponent.getShopService().addItemIdToCatalog(parts[0]);
```
в двух местах используем parts[0]

Предлагаю:
1) сделать метод для парсинга параметров
2) метод для создания предметы из них

после строки String[] parts = newItem.split("[,\\s]+");
вызывать оба метода и иметь предмет Item item = getItemFromParts()

передавать item в addItem
в addItemIdToCatalog передавать item.getId

ptpchh committed:
Potapov/tg_bot/ver_1.7
changed defaultItemInit, added new function getItemFromParts

ptpchh committed:
Potapov/tg_bot/ver_1.7.1
changed defaultItemInit, added new functions getItemFromParts and getPartsFromString
